client/client.go

```diff
@@ -108,10 +111,40 @@ func (client *ThreeScaleClient) doHttpReq(req *http.Request) (ApiResponse, error
 	if err != nil {
 		return authRepRes, err
 	}
+
+	if ext != nil {
+		if _, ok := ext[limitExtensions]; ok {
+			if limitRem := resp.Header[limitRemainingHeaderKey][0]; limitRem != "" {
+				remainingLimit, _ := strconv.Atoi(limitRem)
+				authRepRes.limitRemaining = &remainingLimit
+			}
+
+			if limReset := resp.Header[limitResetHeaderKey][0]; limReset != "" {
+				resetLimit, _ := strconv.Atoi(limReset)
+				authRepRes.limitReset = &resetLimit
```

Comment:
this does not look good if this is meant to be an integer - why not do this:
```golang
authRepRes.limitReset = strconv.Atoi(limReset)
```
(same thing applies just above)

Reply:
Well it returns two values, one of which is an error which we can safely ignore here. Can probably move this to one line. 👍

Reply:
Can we safely ignore it? I mean, if we do, what would be the contents of the int variable? `0`? I think I'd prefer to have it `nil`'ed if so.

Reply:
Well I personally don't like to ignore errors. Bt in this case I thought it was safe to assume that backend will return a value that can be cast to an int meaning we dont have to check the error. However happy to go the more verbose path if you think its safer.

Reply:
Yep, backend won't do that ever, because it's officially bug-free, but you know, sometimes some ops guys tries to modify live production code or the odd contribution ends up affecting backend in the future...

So this is my proposal: don't turn them into pointers, use a struct to contain both ints as normal variables, and use a pointer to the struct to signal it is available. Now, if the parsing fails, you could use a non-sensical value, perhaps a negative value. Wdyt?

Reply:
Done